test/core/test_historical_emailtemplate.py

```diff
@@ -0,0 +1,64 @@
+from django.test import TestCase
+
+
+from squad.core import models
+
+
+class TestHistoricalEmailTemplateTest(TestCase):
+
+    def setUp(self):
+        self.email_template = models.EmailTemplate()
+        self.email_template.name = 'name'
+        self.email_template.subject = 'subject'
+        self.email_template.plain_text = 'plain text'
+        self.email_template.html = 'html'
+        self.email_template.save()
+
+        self.history = self.email_template.history
+
+    def test_history_creation(self):
+        self.assertEqual(len(self.history.all()), 1)
+
+    def test_new_version(self):
+        """
+        This is the main test, for default, any field that has been changed
+        will trigger a new version of the object
+        """
+
+        self.email_template.name = 'other name'
+        self.email_template.save()
+        self.assertEqual(len(self.history.all()), 2)
+
+        self.email_template.subject = 'other subject'
+        self.email_template.save()
+        self.assertEqual(len(self.history.all()), 3)
+
+        self.email_template.plain_text = 'other plain text'
+        self.email_template.save()
+        self.assertEqual(len(self.history.all()), 4)
+
+        self.email_template.html = 'other html'
+        self.email_template.save()
+        self.assertEqual(len(self.history.all()), 5)
+
+        original = self.history.last().instance
+        self.assertEqual(original.name, 'name')
+        self.assertEqual(original.subject, 'subject')
+        self.assertEqual(original.plain_text, 'plain text')
+        self.assertEqual(original.html, 'html')
+
+    def test_revert_to_previous_version(self):
+        self.email_template.subject = 'other subject'
+        self.email_template.save()
+
+        self.email_template = self.history.earliest().instance
+        self.email_template.save()
+        self.assertEqual(len(self.email_template.history.all()), 3)
+        self.assertEqual(self.email_template.name, 'name')
```

Comment:
before testing that the attribute values are back, you need to fetch the object from the database again, otherwise you are just testing the in-memory instance that you already have, and the assestions will pass even if that was not saved to the database (e.g. if you comment the call to `save()`, the assertions about the attributes would still pass)

Reply:
Agreed, I changed that so the model can fetch fresh data and then check if it's been reverted.